### PR TITLE
feat!: Add `secondary_zone` to db modules

### LINKED ISSUE #320: Drift since provider v4.28.0 on google_sql_database_instance.settings.location_preference.secondary_zone
### TL;DR

Since upgrading to `hashicorp/google` v4.28.0, a plan drift appeared that's not fixable when using mssql:
```
~ resource "google_sql_database_instance" "default" {
id                            = "XXX"
name                          = "XXX"
# (14 unchanged attributes hidden)
~settings {
# (10 unchanged attributes hidden)
~location_preference {
-secondary_zone = "us-west1-b" ->null
# (1 unchanged attribute hidden)
            }
# (4 unchanged blocks hidden)
        }
# (1 unchanged block hidden)
    }
Plan: 0 to add, 1 to change, 0 to destroy.
```

- It's not drifting when using a provider older than v4.28.0
- Applying with terraform does not resolve the drift

### Expected behavior

No drift.

### Observed behavior

Drift on `secondary_zone = "us-west1-b" ->null`

### Terraform Configuration

```hcl
module "mssql" {
  source  = "GoogleCloudPlatform/sql-db/google//modules/mssql"
  version = "~> 11.0"
  [...]
}
```
```


### Terraform Version

```sh
Terraform v1.2.4
on darwin_arm64
+ provider registry.terraform.io/cloudflare/cloudflare v3.18.0
+ provider registry.terraform.io/confluentinc/confluent v1.0.0
+ provider registry.terraform.io/elastic/ec v0.4.1
+ provider registry.terraform.io/hashicorp/aws v4.22.0
+ provider registry.terraform.io/hashicorp/azurerm v3.13.0
+ provider registry.terraform.io/hashicorp/external v2.2.2
+ provider registry.terraform.io/hashicorp/google v4.24.0
+ provider registry.terraform.io/hashicorp/google-beta v4.28.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.12.1
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.1.3
+ provider registry.terraform.io/mongodb/mongodbatlas v1.4.2
+ provider registry.terraform.io/strongdm/sdm v2.6.4
```
```


### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jamiezieziula**
Closes #320

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@jamiezieziula
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by reynaldo-rack** [comment]
Good catch @jamiezieziula!

Sorry @bharathkkb do you have any update on this?

I am getting drift because of this...

```
Terraform will perform the following actions:

  # module.sql-db.google_sql_database_instance.default will be updated in-place
  ~ resource "google_sql_database_instance" "default" {
        id                            = "mysqldb-13ae63bd"
        name                          = "mysqldb-13ae63bd"
        # (11 unchanged attributes hidden)

      ~ settings {
            # (10 unchanged attributes hidden)

          ~ location_preference {
              - secondary_zone = "us-central1-a" -> null
                # (1 unchanged attribute hidden)
            }

            # (4 unchanged blocks hidden)
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

**Comment by czka** [comment]
The `secondary_zone` support added by @jamiezieziula's in this PR (thanks!) works fine with my PostgreSQL instance. The PR seems complete to me.

@bharathkkb what's stoping us from merging this? Looks like the Google Cloud Build CI step is broken?

**Comment by jamiezieziula** [comment]
@bharathkkb any update here?

**Comment by jamiezieziula** [comment]
@bharathkkb please let me know what I can do to move this along

**Comment by comment-bot-dev** [comment]
@jamiezieziula
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/mssql/main.tf [hunk 1]
```diff
@@ -122,7 +122,8 @@ resource "google_sql_database_instance" "default" {
     user_labels = var.user_labels
 
     location_preference {
-      zone = var.zone
+      zone           = var.zone
+      secondary_zone = var.secondary_zone
```
Comment (bharathkkb) [inline modules/mssql/main.tf:126]: Looks like this was added in TPG [4.28.0](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md). Can we also bump the required provider versions for all modules? https://github.com/terraform-google-modules/terraform-google-sql-db/blob/95e48a1473714fc5a5ba46b4dce69b21bb6e00d7/modules/mysql/versions.tf#L28-L31

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR and sorry for the delay. A minor comment

File: modules/mssql/main.tf (hunk 1 above)
Comment (jamiezieziula) [reply modules/mssql/main.tf:126]: done!

**Comment by comment-bot-dev** [comment]
@jamiezieziula
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@jamiezieziula
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**